### PR TITLE
Add Contour installed by Operator to kind-e2e test

### PR DESCRIPTION
**Pull request by nak3**
This patch adds contour test installed by operator in kind-e2e test.

net-contour should verify that it works with contour-operator.
Note operator yaml (`./config/contour-operator/`) is not included in release bundle for now.

Fixes https://github.com/knative-sandbox/net-contour/issues/385

### CODE REVIEW

File: hack/tools.go [hunk 1]
```diff
@@ -37,5 +37,6 @@ import (
 	_ "knative.dev/networking/config"
 
 	_ "github.com/mikefarah/yq/v3"
+	_ "github.com/projectcontour/contour-operator/cmd"
```
Comment (dprotaso) [inline hack/tools.go:40]: Do you think we should still be vendoring and building ourselves?

**Comment by dprotaso** [comment]
If so then the operator should probably use the vendored contour

File: hack/tools.go (hunk 1 above)
Comment (dprotaso) [reply hack/tools.go:40]: If so then the operator should probably use the vendored contour

Comment (nak3) [reply hack/tools.go]: I wondered that we should still be building it or not. I think we should not build operator by ourselvs.
Thank you, updated.

**Comment by dprotaso** [comment]
Should we have the operator use our vendored ko version
https://github.com/projectcontour/contour-operator/blob/main/cmd/contour-operator.go#L34-L35

ie. add the flag to on the operator deployment? `--contour-image ko://github.com/projectcontour/contour/cmd/contour`

https://github.com/knative-sandbox/net-contour/blob/eccaac502cac9168c94027a3595b8ad9db26ad23/hack/update-deps.sh#L61-L63

**Comment by nak3** [comment]
Hmm... `ko apply`  command can resolve the image in `--contour-image ko://github.com/projectcontour/contour/cmd/contour` option? I tried but it seems not possible.

**Comment by dprotaso** [comment]
It should work - `ko` checks every string in a yaml file for a `ko://` prefix I believe 

ie. https://github.com/knative/serving/blob/main/config/core/configmaps/deployment.yaml#L27

You would just need to manipulate the operator yaml and add that argument to the container

**Comment by nak3** [comment]
Umm... The configmap works as:

```
$ ko resolve -f config/core/configmaps/deployment.yaml |grep queueSidecarImage
  queueSidecarImage: gcr.io/gcp-compute-engine-223401/queue-39be6f1d08a095bd076a71d288d295b6@sha256:136c0d19dc4094f326b300a20e9737ff0fe8da6b50e1622daa63671469050be1
```

But opertor deployment's command line is not working.

```
$ ko resolve  -f config/contour-operator/operator.yaml  |grep contour-image
            - --contour-image ko://github.com/projectcontour/contour/cmd/contour
```

**Comment by nak3** [comment]
Okay, it seems that the `ko` with the following format can work but it does not work with  `--contour-image` in front of `ko://`.
```yaml
somekey: ko://github.com/projectcontour/contour/cmd/contour
```

**Comment by dprotaso** [comment]
Ah - maybe make it two separate items in the args array?

**Comment by nak3** [comment]
Yeah, the separated line as below works with `ko` but now it hits contour-operator's issue https://github.com/projectcontour/contour-operator/issues/363

```
        - --contour-image 
        - ko://github.com/projectcontour/contour/cmd/contour
```

In short, contour-operator creates label from image tag, but `ko` command creates 64 length tag which exceeds labels length limiation.

**Review by dprotaso — APPROVED** [review]
sounds good - let's revisit when they fix the issue upstream

**Pull request merged**